### PR TITLE
Vend Code Clean

### PR DESCRIPTION
**Pull request by dvir001**
## About the PR
Cleaning the code a bit to remove all the else if with return

## Why / Balance
Clean the code, this isn't a fix for the upstream issue with the search not working

## Technical details
C# Changes

## Media
- [X] This PR does not require an ingame showcase

## Breaking changes
N/A

**Changelog**
N/A

### CODE REVIEW

File: Content.Client/VendingMachines/UI/VendingMachineMenu.xaml.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public VendingMachineMenu()
         /// Populates the list of available items on the vending machine interface
         /// and sets icons based on their prototypes
         /// </summary>
-        public void Populate(List<VendingMachineInventoryEntry> inventory, float priceModifier, string? filter = null)
+        public void Populate(List<VendingMachineInventoryEntry> inventory, string? filter = null, float priceModifier = 10)
```
Comment (Cheackraze) [inline Content.Client/VendingMachines/UI/VendingMachineMenu.xaml.cs:48]: nullable fields should be listed after non-nullables to allow truncating the function calls.  Also, you're bulldozing the price modifier mod by hard-coding it to be permanently 10, discarding any input

File: Content.Client/VendingMachines/VendingMachineBoundUserInterface.cs [hunk 2]
```diff
@@ -42,7 +40,7 @@ protected override void Open()
             _menu.OnItemSelected += OnItemSelected;
             _menu.OnSearchChanged += OnSearchChanged;
 
-            _menu.Populate(_cachedInventory, _mod);
+            _menu.Populate(_cachedInventory, null, _mod);
```
Comment (Cheackraze) [inline Content.Client/VendingMachines/VendingMachineBoundUserInterface.cs:43]: see previous comment. Having your nullable parameters at the end means you dont need to define null here

**Review by Cheackraze — CHANGES_REQUESTED** [review]

**Comment by Cheackraze** [comment]
if im seeing this right, it looks like its an upstream issue about the UI list having outdated cache with the search bar?

File: Content.Client/VendingMachines/UI/VendingMachineMenu.xaml.cs (hunk 1 above)
Comment (dvir001) [reply Content.Client/VendingMachines/UI/VendingMachineMenu.xaml.cs:48]: Didnt notice its not taking the floor from other mod changes, will redo it

**Comment by dvir001** [comment]
> if im seeing this right, it looks like its an upstream issue about the UI list having outdated cache with the search bar?

Yep I reported it, no point in fixing it here unless I fix it on upstream, since its not my code work ill just wait for upstream fix.

**Comment by dvir001** [comment]
> if im seeing this right, it looks like its an upstream issue about the UI list having outdated cache with the search bar?

It was fixed on upstream, ones you merge it ill come back to clean this small code change.

**Comment by Cheackraze** [comment]
> > if im seeing this right, it looks like its an upstream issue about the UI list having outdated cache with the search bar?
> 
> It was fixed on upstream, ones you merge it ill come back to clean this small code change.

thats what i assumed you meant, I will either cherry pick a few fixes or do another upstream next week

**Pull request closed without merging**